### PR TITLE
rector: automatic code refactoring (part 1)

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

We had a lot of code-style improvements made manually (or with help of phpstorm in my case). E.g. Tenery operators. A lot of changes were only made when the file has been touched for some other reason.

Why not automate this and fix all occourences?

I had time to test [rector](https://github.com/rectorphp/rector) and i am positively surprised!

I tried ~50 rules step by step and ... there is not _soo_ much to change. (stopped with ~500 changed files ... incl. lib)

1. finish [rulset](https://github.com/sreichel/openmage-rector/blob/main/rector.php)
2. apply changes in small steps
3. add OM specific rules (to update 3rd-party-code)

### Manual testing scenarios (*)

1. `composer require --dev sreichel/openmage-rector`
4. `cp -n vendor/sreichel/openmage-rector/rector.php rector.php`
5. `php vendor/bin/rector process <path-to-files> --dry-run`

Note: running w/o path (whole project) takes _some_ memory.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

**Comment by elidrissidev** [comment]
Cool stuff. But why are certain CI workflows being skipped?

**Review by fballiano — APPROVED** [review]
beautiful

**Comment by sreichel** [comment]
> Cool stuff. But why are certain CI workflows being skipped?

Some checks run on schedule at the moment ... https://github.com/OpenMage/magento-lts/actions?query=event%3Aschedule

**Review by elidrissidev — APPROVED** [review]

File: app/code/core/Mage/Catalog/Model/Product/Attribute/Backend/Startdate.php [hunk 1]
```diff
@@ -39,9 +39,6 @@ protected function _getValueForSave($object)
     {
         $attributeName  = $this->getAttribute()->getName();
         $startDate      = $object->getData($attributeName);
-        if ($startDate === false) {
```
Comment (tmotyl) [inline app/code/core/Mage/Catalog/Model/Product/Attribute/Backend/Startdate.php:42]: Is this change safe? It sounds like it relies on the doctype, which might not be correct.

Comment (sreichel) [reply app/code/core/Mage/Catalog/Model/Product/Attribute/Backend/Startdate.php:42]: It is. Code before ...

```
        if ($startDate === false) {
            return false;
        }
        return $startDate;
```

This check is useles.

**Pull request merged**